enhancements/network/user-defined-network-for-virt-workloads.md

```diff
@@ -0,0 +1,604 @@
+meaning KubeVirt (or a component on behalf of it) will create the `IPAMClaim`
+and will instruct OVN-Kubernetes of which claim to use. Since for primary UDNs
+we do **not** define the network-selection-elements, we need to use a new
+annotation to pass this information along.
+
+The proposed annotation value is `k8s.ovn.org/ovn-udn-ipamclaim-reference`.
+
+These persistent IPs will be cleaned up by the Kubernetes garbage collector
+once the VM to which they belong is removed.
+
+All the other
+[work required in OVN-Kubernetes](https://github.com/trozet/enhancements/blob/941f5c6391830d5e4a94e65d742acbcaf9b8eda9/enhancements/network/user-defined-network-segmentation.md#pod-egress)
+to support live-migration was already implemented as part of the epic
+implementing UDN.
+
+### Allow user to configure the VMs interface desired IP address
+
+We will also rely on the `IPAMClaim` CRD to provide the VM owner a way to
+specify the IP addresses for the UDN interface of their VMs. This way, we
+do not require to update the KubeVirt API, which is something we have faced
+extreme resistance in the past.
+
+Keep in mind that up to now the IPAMClaims would be created by KubeVirt
+directly; we need to allow the VM user to create them, specifying in its `spec`
+the desired IP addresses for the VM, and when creating the VM, point it to the
+respective `IPAMClaim` using an annotation (on the VM).
+
+KubeVirt (or a component on behalf of it) will see the annotation on the VM,
+and will proceed to the `IPAMClaim` owner reference, and template the launcher
+pod accordingly - with the annotation mentioned in the
+[Persisting VM IP addresses during the migration](#persisting-vm-ip-addresses-during-the-migration)
+section.
+
+Once OVN-Kubernetes reconciles the pod which will host the VM, it will first
+validate the request - i.e. are the requested IPs in the configured ranges of
+the UDN network ? If not, the `IPAMClaim` condition will be set accordingly,
+and allocating the OVN annotation for the pod will fail. The network controller
+will keep attempting to reconcile the pod using exponential backoff until it
+possibly succeeds - in the case the user updates the
+`IPAMClaim.Spec.IPRequests` attribute.
+
+If the IPs being requested are in the configured subnet range of the network,
+the OVN-Kuberntes network controller proceeds to allocate the requested IP
+addresses; it might fail in case the requested IP addresses are already
+allocated in the network. If so, the pod will crashloop, and the `IPAMClaim`
+conditions updated to reflect this.
+
+In the case it succeeds, the `IPAMClaim` conditions are updated w/ a success
+condition, and its `IPAMCLaim.Status.IPs` updated accordingly (as happens today)
+for secondary networks.
+
+### Workflow Description
+
+#### Pre-requirements - configure the primary UDN for a namespace
+The user will need to perform the usual steps to configure a primary UDN for
+their namespace - that is, to provision a `UserDefinedNetwork` of layer2
```

Comment:
Does primary-UDN only support layer2 topology? I remember the UDN enhancement talking also on layer3..

Reply:
for the virt integration, we only care about L2.